### PR TITLE
Windows / AppVeyor support

### PR DESCRIPTION
**Pull request by photex**
Howdy!

As discussed in #37 I'm proposing the following changes to support windows, and I've added an appveyor config as requested.

### CODE REVIEW

**Comment by photex** [comment]
I'm experimenting with appveyor at the moment...

**Comment by photex** [comment]
Okidoki I think that's going to do it for appveyor. The gnu toolchain won't be able to build this yet without some tweaks. lmdb sources are trying to use the win32 apis still, which aren't being found, and I don't have the bandwidth right this second to figure that whole thing out.

This appears to work great for msvc when building for 32-bit and 64-bit targets.

Sometime down the road when I have the opportunity I'll probably try and work out what needs to happen to support the gnu toolchain as well.

**Comment by photex** [comment]
Just realized I was including a useless merge commit.. just force pushed and removed that.

File: liblmdb-sys/src/lib.rs [hunk 1]
```diff
@@ -8,23 +8,22 @@ use libc::{c_int, c_uint, c_void, c_char, size_t};
 #[cfg(any(target_os = "macos", target_os = "ios", target_os = "linux",
           target_os = "freebsd", target_os = "dragonfly",
           target_os = "android"))]
-mod os {
+pub mod os {
```
Comment (vhbit) [inline liblmdb-sys/src/lib.rs]: I don't think you need to make this `mod` public - re-export of `mdb_mode_t` actually happens a bit higher in this file https://github.com/vhbit/lmdb-rs/blob/master/liblmdb-sys/src/lib.rs#L5

File: liblmdb-sys/src/lib.rs [hunk 2]
```diff
@@ -8,23 +8,22 @@ use libc::{c_int, c_uint, c_void, c_char, size_t};
...
     pub type mdb_mode_t = libc::mode_t;
-    
+
     pub type mdb_filehandle_t = libc::c_int;
 }
 
 #[cfg(target_os = "windows")]
-mod os {
-    use libc::{c_int, c_void};
+pub mod os {
```
Comment (vhbit) [inline liblmdb-sys/src/lib.rs]: Same as above.

File: src/core.rs [hunk 3]
```diff
@@ -643,7 +643,7 @@ impl EnvBuilder {
             let c_path = try!(CString::new(path_str).map_err(|_| MdbError::InvalidPath));
 
             ffi::mdb_env_open(mem::transmute(env), c_path.as_ptr(), self.flags.bits(),
-                              perms as libc::mode_t)
+                              perms as ffi::os::mdb_mode_t)
```
Comment (vhbit) [inline src/core.rs]: and considering changes above it should be `ffi::mdb_mode_t`

**Comment by photex** [comment]
okidoki, updated with your comments addressed.

**Comment by vhbit** [comment]
Thanks!

**Pull request merged**